ts/features/idpay/configuration/components/IdPayDiscountInstrumentEnrollmentSwitch.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  IOIcons,
+  ListItemSwitch,
+  SwitchAction
+} from "@pagopa/io-app-design-system";
+import { default as React } from "react";
+import {
+  InstrumentDTO,
+  StatusEnum
+} from "../../../../../definitions/idpay/InstrumentDTO";
+import { IDPayDiscountInitiativeInstruments } from "../types";
+import I18n from "../../../../i18n";
+
+export type PaymentMethodSwitchRef = {
+  switchStatus: boolean;
+  setSwitchStatus: (status: boolean) => void;
+};
+
+type PaymentMethodSwitchProps = {
+  instrumentPaymentMethod: InstrumentDTO;
+  onValueChange: (
+    paymentType: IDPayDiscountInitiativeInstruments,
+    value: boolean
+  ) => void;
+  onPressAction?: (paymentType: IDPayDiscountInitiativeInstruments) => void;
+};
+
+const getInstrumentPaymentMethodIcon = (
+  paymentType: IDPayDiscountInitiativeInstruments
+): IOIcons => {
+  switch (paymentType) {
+    case IDPayDiscountInitiativeInstruments.APP_IO:
+      return "device";
+    case IDPayDiscountInitiativeInstruments.CIE:
+      return "fiscalCodeIndividual";
+  }
+};
+
+const getPaymentTypeFromInstrument = (
+  instrumentPaymentMethod: InstrumentDTO
+) => {
+  switch (instrumentPaymentMethod.brand) {
```

Comment:
`InstrumentDTO` comes with `instrumentType`, which indicates if the instrument is a card (`CARD`), QrCode (`QRCODE`) and CIE (`IDPAYCODE`). We could use this instead of the mapped brand with `IDPayDiscountInitiativeInstruments`. What do you think?

Reply:
Yep, this PR was not yet integrated with the new definition of the swagger